python-frontend/src/main/java/org/sonar/python/types/v2/ModuleType.java

```diff
@@ -36,6 +36,14 @@ public ModuleType(@Nullable String name, @Nullable ModuleType parent, Map<String
     this.name = name;
     this.parent = parent;
     this.members = members;
+    if (this.parent == null) {
+      return;
```

Comment:
having an early return in a simple constructor confused me a bit from understanding what's going on here. Maybe you can extend this logic to a private method like `registerToParent` or revert the condition